### PR TITLE
chore(monorepo): remove unnecessary build tooling from plugins; simplify peer-deps check CLOUDP-207383

### PR DESCRIPTION
**Pull request by gribnoysup**
This is a follow-up patch to "vendoring" the monorepo packages that cloud is using. As cloud will not be using compass packages directly, we can simplify some checks and compilation steps for plugins:

- we don't need to compile plugins with webpack anymore, so plugin webpack config is removed and all plugins are switched to use tsc instead (we still need to for type def generation), this should speed up bootstrap / compile and removes some unneccessary differences between workspace templates
- we don't need to maintain special rules for any dependencies groups in package.json in monorepo packages, so the check is simplified to: everything used in source (and so being used in dist) needs to be in prod dependencies, everything else is dev dependencies, peer dependencies are not allowed (this is aligned with the default npm behavior that will install peer deps anyway). The `check-peer-deps` script it updated to validate that and also has an `--autofix` option now. It doesn't validate unused dependencies or dependencies missing outside of source, we are still using `depcheck` for this (although we can replace it and remove a few hacks we are keeping around for it to work if we want to)

Doing this alignment already caught some issues where we had dependencies not actually used by packages at all (seems like having something both in prod and peer deps was silently tripping `depcheck` package before) or places where we were referencing non-existent files in source, so this patch also addresses these issues.

For reviewing, the third commit here (c8aa2141016071922a8d428a9d667eec58745c45) is autogenerated with a script and is the biggest one, I'm pretty sure it should be okay to just give it a glance as it's just running autofix on the packages and auto-cleans up some config files to align with the updated `create-workspace` script. Everything else I tried to group into meaningful small commits that should be easy enough to review separately

### CODE REVIEW

File: packages/instance-model/index.d.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import type Collection from 'mongodb-collection-model';
 import type { DataService } from 'mongodb-data-service';
 import type { Collection as DatabaseCollection } from 'mongodb-database-model';
-
-import { ServerType } from './server-type';
-import { TopologyType } from './topology-type';
 import Database from 'mongodb-database-model';
 import { CollectionCollection } from 'mongodb-collection-model';
 
+// TODO: add real types for these
```
Comment (lerouxb) [inline packages/instance-model/index.d.ts:7]: nit: should we add a ticket?

File: packages/compass-app-stores/package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
   },
   "types": "./dist/index.d.ts",
   "scripts": {
-    "bootstrap": "npm run postcompile",
```
Comment (lerouxb) [inline packages/compass-app-stores/package.json:37]: For a moment I couldn't figure out how this worked before then I realised postcompile was already removed in a previous commit 👍

**Review by lerouxb — APPROVED** [review]

**Review by mcasimir — APPROVED** [review]

File: packages/instance-model/index.d.ts (hunk 1 above)
Comment (gribnoysup) [reply packages/instance-model/index.d.ts:7]: I'll open a follow-up right away! Didn't want to make the amount of required changes here bigger than needed

File: packages/compass-app-stores/package.json (hunk 2 above)
Comment (gribnoysup) [reply packages/compass-app-stores/package.json:37]: Yeah, sorry, this one should've been part of automated changes, but I messed up the script a bit, didn't want to re-run the whole thing, and so had to split it to a separate commit 🙈

**Pull request merged**